### PR TITLE
Test/uenv tests

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

**Comment by teojgo** [comment]
cscs-ci run

File: checks/uenv/gromacs_check.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Copyright 2016-2022 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+from hpctestlib.sciapps.gromacs.benchmarks import gromacs_check
```
Comment (jgphpc) [inline checks/uenv/gromacs_check.py:7]: No module named ['hpctestlib'](https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/6557018579987861/835515875116199/-/jobs/4138956488#L92)

File: checks/uenv/check_osu_micro_benchmarks_uenv.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    num_tasks = 2
+    num_tasks_per_node = 1
+    modules = ['osu-micro-benchmarks']
+    pmi = variable(str, value='')
+    env_vars = {
+        'MPIR_CVAR_ENABLE_GPU': 0,
+        # Set to one byte more than the last entry of the test
+        'MPIR_CVAR_CH4_OFI_MULTI_NIC_STRIPING_THRESHOLD': 4194305
+    }
+    maintainers = ['TM']
```
Comment (jgphpc) [inline checks/uenv/check_osu_micro_benchmarks_uenv.py:18]:
```suggestion
    maintainers = ['@teojgo']
```

**Comment by teojgo** [comment]
cscs-ci run

**Comment by teojgo** [comment]
cscs-ci run

**Comment by teojgo** [comment]
cscs-ci run default

File: checks/uenv/check_osu_micro_benchmarks.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    tags = {'uenv'}
+
+    @run_after('init')
+    def set_descr(self):
+        self.descr = f'{self.name} on {self.num_tasks} nodes(s)'
+
+    @run_after('setup')
+    def set_launcher_options(self):
+        if self.pmi != '':
+            self.job.launcher.options = [f'--mpi={self.pmi}']
```
Comment (jgphpc) [inline checks/uenv/check_osu_micro_benchmarks.py:33]:
```suggestion
            self.job.launcher.options = \[self.current_environ.extras['launcher_options']]
```
as in https://github.com/eth-cscs/cscs-reframe-tests/blob/alps/config/systems/hohgant.py#L191

**Review by jgphpc — APPROVED** [review]

**Review by ekouts — APPROVED** [review]

**Pull request merged**